streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1000,7 +1012,17 @@ public void restore(final Map<TaskId, Task> tasks) {
             CLIENT_ID,
             new LogContext(""),
             new AtomicInteger(),
-            new AtomicLong(Long.MAX_VALUE)
+            new AtomicLong(Long.MAX_VALUE),
+            null,
+            e -> {
+                if (e instanceof RuntimeException) {
+                    throw (RuntimeException) e;
+                } else if (e instanceof Error) {
+                    throw (Error) e;
+                } else {
+                    throw new RuntimeException("Unexpected checked exception caught in the uncaught exception handler", e);
+                }
```

Comment:
Why set the exception handler in this test and no others?

Reply:
Because otherwise the task migrated exception sends it into a endless rebalance

Reply:
But TaskMigratedException should never be thrown all the way up to the exception handler. Is that what you're seeing?

Reply:
Not quite. If I remove the handler and just run it there is an illegal state exception which runs endlessly until the handler can exit the loop. It looks like the thread hadn't started all the way before the TaskMigratedExcpetion is thrown

`INFO State transition from STARTING to PENDING_SHUTDOWN (org.apache.kafka.streams.processor.internals.StreamThread:223)
[`

Reply:
Ah ok so there's some other IllegalStateException that would get swallowed if we just used `e -> {}` like in the other tests, so we need to explicitly rethrow it? That seems fine, although it makes me think that we should go ahead and use a "real" handler in _all_ of the tests, not just this one. Otherwise there could be some bug which causes an unexpected exception, but the test would just swallow it and silently pass.
Can we just use the default handler wrapper for all of these tests so they reflect realistic scenarios?

Reply:
The default is in KafkaStreams, but I see your point. We can make all of them rethrow then we will not have to worry about swallowing